client/state/sites/selectors.js

```diff
@@ -550,3 +553,311 @@ export function getSitePostsPage( state, siteId ) {
+ * @param {Number} siteId Site ID
+ * @return {Boolean} if the site is a secondary network site
+ */
+export function isJetpackSiteSecondaryNetworkSite( state, siteId ) {
+	const site = getRawSite( state, siteId );
+
+	if ( ! site ) {
+		return null;
+	}
+
+	const isMultiNetwork = getSiteOption( state, siteId, 'is_multi_network' );
+
+	if ( site.is_multisite === false || isMultiNetwork === false ) {
+		return false;
+	}
+
+	const unmappedUrl = getSiteOption( state, siteId, 'unmapped_url' ),
+		mainNetworkSite = getSiteOption( state, siteId, 'main_network_site' );
+
+	// Compare unmapped_url with the main_network_site to see if is not the main network site.
+	return withoutHttp( unmappedUrl ) !== withoutHttp( mainNetworkSite );
+}
+
+/**
+ * Determines if all given modules are active for a Jetpack Site
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @param {Array} moduleIds A list of active module ids to verify
+ * @return {Boolean} if the all the given modules are active for this site
+ */
+export function verifyJetpackModulesActive( state, siteId, moduleIds ) {
+	if ( ! isJetpackSite( state, siteId ) ) {
+		return null;
+	}
+
+	if ( ! Array.isArray( moduleIds ) ) {
+		moduleIds = [ moduleIds ];
+	}
+
+	return every( moduleIds, moduleId => isJetpackModuleActive( state, siteId, moduleId ) );
+}
+
+/**
+ * Returns the remote management url for a Jetpack site
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {String} the remote management url for the site
+ */
+export function getJetpackSiteRemoteManagementURL( state, siteId ) {
+	if ( ! isJetpackSite( state, siteId ) ) {
+		return null;
+	}
+
+	const siteJetpackVersion = getSiteOption( state, siteId, 'jetpack_version' ),
+		siteAdminUrl = getSiteOption( state, siteId, 'admin_url' ),
+		configure = versionCompare( siteJetpackVersion, '3.4' ) < 0 ? 'manage' : 'json-api';
+
+	return siteAdminUrl + 'admin.php?page=jetpack&configure=' + configure;
+}
+
+/**
+ * Checks whether a Jetpack site has a custom mapped URL.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {?Boolean} Whether site has custom domain
+ */
+export function hasJetpackSiteCustomDomain( state, siteId ) {
+	const site = getRawSite( state, siteId ),
+		domain = getSiteDomain( state, siteId ),
+		unmappedUrl = getSiteOption( state, siteId, 'unmapped_url' );
+	if ( ! site || ! domain || ! unmappedUrl ) {
+		return null;
+	}
+
+	return domain !== withoutHttp( unmappedUrl );
+}
+
+/**
+ * Returns an explanation on why updates are disabled on a Jetpack Site
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @param {String} action The update action we wanted to perform on this site
+ * @return {?String} The reason why file update is disabled
+ */
+export function getJetpackSiteFileModDisableReasons( state, siteId, action = 'modifyFiles' ) {
```

Comment:
Should we update this name to `getJetpackSiteUpdateFilesDisabledReasons` to align with `canJetpackSiteUpdateFiles`? Any other ideas how to rename `Mod` part?